src/wallet/rpcwallet.cpp

```diff
@@ -538,7 +538,7 @@ UniValue signmessage(const UniValue& params, bool fHelp)
 
     vector<unsigned char> vchSig;
     if (!key.SignCompact(ss.GetHash(), vchSig))
-        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Sign failed");
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Sign failed");
```

Comment:
This should be something different than RPC_INVALID_PARAMETER (maybe a general "internal error" or introduce a new RPC_SIGN_FAILED).